src/main/scala/is/hail/utils/HashMethods.scala

```diff
@@ -32,3 +32,51 @@ class TwoIndepHash32(outBits: Int, a: Long, b: Long) extends (Int => Int) {
 
   override def apply(key: Int): Int = ((a * key + b) >>> shift).toInt
 }
+
+object SimpleTabulationHash32 {
+  def apply(rand: RandomDataGenerator): SimpleTabulationHash32 = {
+    new SimpleTabulationHash32(Array.fill[Array[Int]](4)(Array.fill[Int](256)(rand.getRandomGenerator.nextInt())))
```

Comment:
You didn't flatten this one?

Reply:
Oops, sorry! I got turned around jumping between branches.